### PR TITLE
Remove some compiler warnings

### PR DESCRIPTION
**Pull request by ygator**
Removed compiler warnings.

### CODE REVIEW

**Comment by merbanan** [comment]
The last hunk looks ok. The others Im not sure about.

**Comment by ygator** [comment]
data.c: In function ‘print_kv_data’:
data.c:426:35: warning: zero-length gnu_printf format string [-Wformat-zero-length]
                     fprintf(file, "");

rtl_433.c:  In function ‘main’:
rtl_433.c:1197:12: warning: ‘r’ may be used uninitialized in this function [-Wmaybe-uninitialized]
     return r >= 0 ? r : -r;

devices/x10_rf.c: In function ‘X10_RF_callback’:
devices/x10_rf.c:20:16: warning: comparison of promoted ~unsigned with unsigned [-Wsign-compare]
   && (bb[1][2] == (uint8_t)(~bb[1][3]))  // Check integrity


fprintf(file, "") does not output anything so replace it with nothing so it does not produce warning.

~bb[1][3] is equivalent to 255-bb[1][3] and does not produce warning.

File: src/data.c [hunk 1]
```diff
@@ -423,7 +423,7 @@ static void print_kv_data(data_printer_context_t *printer_ctx, data_t *data, cha
 			}
 		}
 		if (!strcmp(data->key, "time"))
-                    fprintf(file, "");
+                    ;
```
Comment (merbanan) [inline src/data.c]: Change this to a commented fprintf statement. Otherwise the code looks very peculiar.

File: src/devices/x10_rf.c [hunk 2]
```diff
@@ -17,7 +17,7 @@ static int X10_RF_callback(bitbuffer_t *bitbuffer) {
 	// Validate package
 	if ((bitbuffer->bits_per_row[1] == 32)		// Dont waste time on a short package
 	// && (bb[1][0] == (uint8_t)(~bb[1][1]))		// Check integrity - apparently some chips may use both bytes..
-	 && (bb[1][2] == (uint8_t)(~bb[1][3]))		// Check integrity
+	 && (bb[1][2] == (255 - bb[1][3]))		// Check integrity
 	)
```
Comment (merbanan) [inline src/devices/x10_rf.c]: Can you cast the bb[1][2] to (uint8)? Doing 255 - instead is very obfuscated.

File: src/rtl_433.c [hunk 3]
```diff
@@ -797,7 +797,7 @@ int main(int argc, char **argv) {
     char *in_filename = NULL;
     FILE *in_file;
     int n_read;
-    int r, opt;
+    int r = 0, opt;
     int i, gain = 0;
```
Comment (merbanan) [inline src/rtl_433.c]: This is ok.

File: src/devices/x10_rf.c (hunk 2 above)
Comment (ygator) [inline src/devices/x10_rf.c]: I had tried different casts, but it did not like any of them.
A separate variable was the only way i.e.
uint8_t x = ~bb[1][3];
bb[1][2] == x
Then I though well the ~ of a byte is the same as 255 - byte and that ended up not generating a compiler warning.

File: src/data.c (hunk 1 above)
Comment (ygator) [inline src/data.c]: done

**Comment by ygator** [comment]
I added a comment for the 255 - code.

**Comment by zuckschwerdt** [comment]
It seems to be a strange (and old) bug in GCC. https://gcc.gnu.org/bugzilla/show_bug.cgi?id=38341
An interesting read and if I follow correctly ~x won't be the same as 255-x because of promotion to int -- that shouldn't apply with an explicit cast though.

**Comment by ygator** [comment]
If x is 0x81 then we get:
~x = ffffff7e
255 -x = 7e

Casting ~x to uint8_t still gives compiler warning.
So 255 - x is the cleanest solution I see.

**Comment by ygator** [comment]
I am not sure who original wrote the code, but perhaps they did not get the warning at all.  I am on archlinuxarm and got those warnings.

**Comment by zuckschwerdt** [comment]
We know it's a GCC bug (no warning with clang) and the code works as intended. I'd opt for a GCC pragma:
```
#pragma GCC diagnostic ignored "-Wsign-compare"
```

**Comment by merbanan** [comment]
Jolly good.

**Pull request merged**